### DISCUSSION TITLE
Unknown argument `id`. Available options are marked with ?.

### DISCUSSION THREAD

**Discussion by apio-monitor** [prisma/prisma#23324] [Q&A] [answered]
### Question

Getting error mentioned in title. my code is below

**Schema.prisma**
```prisma
model Survey {
  id          Int     @id @default(autoincrement())
  title       String
  description String
  shop        String   @default("")
  surveyquestions SurveyQuestion[]
  responses   SurveyResponse[] 
  createdAt   DateTime @default(now())
  updatedAt   DateTime  @updatedAt
}

model Question {
  id          Int     @id @default(autoincrement())
  title       String
  description String
  type        String ///should be enum type but enum is not supported in sqlite
  surveyquestions SurveyQuestion[]
  options     Option[]
  createdAt   DateTime @default(now())
  updatedAt   DateTime  @updatedAt
}

model SurveyQuestion{
  id          Int     @id @default(autoincrement())
  survey      Survey  @relation(fields: [surveyId], references: [id])
  surveyId    Int 
  question    Question @relation(fields: [questionId], references: [id])
  questionId  Int
  responses   SurveyResponse[]
  createdAt   DateTime @default(now())
  updatedAt   DateTime  @updatedAt
}
```
**My router code(im using remix.run)**
```js
export async function action ({ request, params }){
    let formData = await request.formData();
    let values  = Object.fromEntries(formData);
    console.log(values)
    let survey_id = Number(params.surveyId);
    survey_obj = await prisma.survey.findUnique({
      where: {
        id: survey_id,
      },
    });
    console.log(survey_id)
    question_obj = await prisma.question.findUnique({
      where: {
        id: 10,
      },
    }) 
    // dummy for now
    console.log(values);
    
    const surveyquestion = await prisma.SurveyQuestion.create({
      data: {
        survey: survey_obj,
        question: question_obj,
      },
    })
    return null;
  }
```
----
What am I doing wrong ?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @apio-monitor 👋 

Thank you for raising this question.

Can you please share the complete error log?

**Comment by apio-monitor** [comment]
hi @ludralph below is the complete error log --

PrismaClientValidationError: 
Invalid `prisma.SurveyQuestion.create()` invocation in
/Users/nikhilmanhas/Desktop/survey-app-l21/app/routes/app.add_questions.$surveyId.jsx:87:56

  84 // dummy for now
  85 console.log(values);
  86 
→ 87 const surveyquestion = await prisma.SurveyQuestion.create({
       data: {
         survey: {
           id: 73,
           ~~
           title: "rere",
           description: "rerer",
           shop: "",
           createdAt: new Date("2024-03-01T15:19:18.414Z"),
           updatedAt: new Date("2024-03-01T15:19:18.414Z"),
     ?     create?: SurveyCreateWithoutSurveyquestionsInput | SurveyUncheckedCreateWithoutSurveyquestionsInput,
     ?     connectOrCreate?: SurveyCreateOrConnectWithoutSurveyquestionsInput,
     ?     connect?: SurveyWhereUniqueInput
         },
         question: {
           id: 10,
           title: "edede",
           description: "deeddede",
           type: "text",
           createdAt: new Date("2024-02-20T09:15:40.191Z"),
           updatedAt: new Date("2024-02-20T09:15:40.191Z")
         }
       }
     })

Unknown argument `id`. Available options are marked with ?.
    at ri (/Users/nikhilmanhas/Desktop/survey-app-l21/node_modules/@prisma/client/runtime/library.js:119:5888)
    at ai.handleRequestError (/Users/nikhilmanhas/Desktop/survey-app-l21/node_modules/@prisma/client/runtime/library.js:126:6431)
    at ai.handleAndLogRequestError (/Users/nikhilmanhas/Desktop/survey-app-l21/node_modules/@prisma/client/runtime/library.js:126:6109)
    at ai.request (/Users/nikhilmanhas/Desktop/survey-app-l21/node_modules/@prisma/client/runtime/library.js:126:5817)
    at l (/Users/nikhilmanhas/Desktop/survey-app-l21/node_modules/@prisma/client/runtime/library.js:131:9709)
    at action (/Users/nikhilmanhas/Desktop/survey-app-l21/app/routes/app.add_questions.$surveyId.jsx:87:28)
    at Object.callRouteActionRR (/Users/nikhilmanhas/Desktop/survey-app-l21/node_modules/@remix-run/server-runtime/dist/data.js:35:16)
    at callLoaderOrAction (/Users/nikhilmanhas/Desktop/survey-app-l21/node_modules/@remix-run/router/router.ts:4020:16)
    at submit (/Users/nikhilmanhas/Desktop/survey-app-l21/node_modules/@remix-run/router/router.ts:3133:16)
    at queryImpl (/Users/nikhilmanhas/Desktop/survey-app-l21/node_modules/@remix-run/router/router.ts:3068:22)

**Comment by apio-monitor** [comment, marked as answer]
got it actually changing to below code in route worked.. since surveyId, questionId are the fields in the database not survey, question.. thanks for help .. 
```
const surveyquestion = await prisma.SurveyQuestion.create({
      data: {
        surveyId: survey_obj.id,
        questionId: question_obj.id,
      },
    })
```
